tests/bluetooth/bsim_bt/bsim_test_mesh/src/distribute_keyid.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/bluetooth/mesh.h>
+#include "argparse.h"
+#include "mesh/crypto.h"
+
+#define LOG_MODULE_NAME distribute_keys
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Mesh requires to keep in persistent memory network keys (2 keys per subnetwork),
+ * application keys (2 real keys per 1 configured) and device key.
+ */
+#define KEY_ID_RANGE_SIZE (2 * CONFIG_BT_MESH_SUBNET_COUNT + \
+		2 * CONFIG_BT_MESH_APP_KEY_COUNT + 1)
+
+static psa_key_id_t pst_key_id[KEY_ID_RANGE_SIZE] = {PSA_KEY_ID_NULL};
+
+psa_key_id_t bt_mesh_user_keyid_alloc(void)
+{
+	unsigned int unique = get_device_nbr();
+
+	unique <<= 8;
+
+	for (int i = 0; i < KEY_ID_RANGE_SIZE; i++) {
+		if (pst_key_id[i] == PSA_KEY_ID_NULL) {
+			pst_key_id[i] = PSA_KEY_ID_USER_MIN + i + unique;
+			LOG_INF("key id %d is allocated", pst_key_id[i]);
+			return pst_key_id[i];
+		}
+	}
+
+	return PSA_KEY_ID_NULL;
+}
+
+int bt_mesh_user_keyid_free(psa_key_id_t key_id)
+{
+	int err = -EIO;
+	unsigned int unique = get_device_nbr();
+
+	unique <<= 8;
+
+	if (IN_RANGE(key_id - unique, PSA_KEY_ID_USER_MIN,
+			PSA_KEY_ID_USER_MIN + KEY_ID_RANGE_SIZE - 1)) {
+		LOG_INF("key id %d is freed", pst_key_id[key_id - PSA_KEY_ID_USER_MIN]);
+		pst_key_id[key_id - PSA_KEY_ID_USER_MIN] = PSA_KEY_ID_NULL;
+		err = 0;
+	}
+
+	return err;
```

Comment:
Suggest to move successful case out of `if`:
```suggestion
	unsigned int unique = get_device_nbr();

	unique <<= 8;

	if (!IN_RANGE(key_id - unique, PSA_KEY_ID_USER_MIN,
			PSA_KEY_ID_USER_MIN + KEY_ID_RANGE_SIZE - 1)) {
		return -EIO;
	}
	
	LOG_INF("key id %d is freed", pst_key_id[key_id - PSA_KEY_ID_USER_MIN]);
	pst_key_id[key_id - PSA_KEY_ID_USER_MIN] = PSA_KEY_ID_NULL;

	return 0;
```